components/core/tests/test-kql.cpp

```diff
@@ -0,0 +1,114 @@
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include <Catch2/single_include/catch2/catch.hpp>
+
+#include "../src/clp_s/search/AndExpr.hpp"
+#include "../src/clp_s/search/FilterExpr.hpp"
+#include "../src/clp_s/search/kql/kql.hpp"
+#include "../src/clp_s/search/OrExpr.hpp"
+
+using clp_s::search::AndExpr;
+using clp_s::search::DescriptorToken;
+using clp_s::search::FilterExpr;
+using clp_s::search::FilterOperation;
+using clp_s::search::kql::parse_kql_expression;
+using clp_s::search::OrExpr;
+using std::string;
+using std::stringstream;
+using std::vector;
+
+TEST_CASE("Test parsing KQL", "[KQL]") {
+    // Initialize data for testing
+    vector<string> const basic_equivalent_wildcard_queries{
+            "value",
+            "\"value\"",
+            "*:value",
+            "*:\"value\"",
+            "\"*\":value"
+    };
+
+    SECTION("Parsing basic wildcard column queries") {
+        for (auto const& query : basic_equivalent_wildcard_queries) {
+            stringstream wildcard_filter{query};
+            std::string extracted_value;
+            auto filter
+                    = std::dynamic_pointer_cast<FilterExpr>(parse_kql_expression(wildcard_filter));
+            REQUIRE(nullptr != filter);
+            REQUIRE(false == filter->is_inverted());
+            REQUIRE(filter->get_column()->is_pure_wildcard());
+            REQUIRE(FilterOperation::EQ == filter->get_operation());
+            REQUIRE(filter->get_operand()->as_var_string(extracted_value, filter->get_operation()));
+            REQUIRE("value" == extracted_value);
```

Comment:
Do you consider adding `REQUIRE(DescriptorToken{"*"} == *filter->get_column()->descriptor_begin());`?

Reply:
What was the conclusion here?

Reply:
I don't think it makes sense since "is_pure_wildcard" is the only way that code should check for a pure wildcard column (and its the only way that code does check for this condition). Checking for descriptor token '*' also starts to feel wrong when a user enters something like "*.*" as their column, which should also be a pure wildcard. (Granted, repeated wildcards in a column descriptor get sanitized and turned into single wildcards to simplify the column resolution code in my other PR).

Reply:
Gotcha, thanks